drivers/console/aarch64/console.S

```diff
@@ -95,10 +95,15 @@ endfunc console_getc
 	 * data that hasn't been output. It returns 0
 	 * upon successful completion, otherwise it
 	 * returns -1.
-	 * Clobber list : x0, x1
+	 * Clobber list : x0, x1, x2, x3, x4
 	 * ---------------------------------------------
 	 */
 func console_flush
+#if COREBOOT
+	mov	x4, x30
+	bl	coreboot_cbmc_flush
+	mov	x30, x4
+#endif
```

Comment:
I'm just trying to come up with a system that doesn't require every driver to essentially be written twice. If drivers implement a new API in a separate file as you suggested, I'm afraid it would create a mess of drivers that only support one API but not the other depending on what the implementer needed. I think it would be nicer if we can rewrite the existing drivers to a framework that can support both APIs at once, so that future driver additions will also support both automatically.

I'll try to put this into code and see how it works out in practice.